### DISCUSSION TITLE
some question about stop animation

### DISCUSSION THREAD

**Discussion by simonsland** [google/filament#5436] [Q&A] [answered]
I fount that animator only provides applyAnimation() to start animation, but there is no interface to stop animation. It's not convenient to back model to T-pose when the frames of animation  don't include the T-pose. Is there any good solutions to do solve this problem?

**Comment by bejado** [comment]
`applyAnimation()` should be called every frame to update the animation (note how it takes a `float time`). So to "stop animation", you'd simply stop calling `applyAnimation()`.

**Reply by simonsland** [reply]
yes, stop calling `applyAnimation()` can "stop animation". But the model won't back to the T-pose(init state before play animation), it just stoped at the frame of when we stop calling `applyAnimation()`.

**Reply by romainguy** [reply]
What's missing is a way to reset the nodes' transform and morph weights for a given animation. This can be done by hand technically but it's awkward. Sounds like we should add a `resetAnimation(animationIndex)` to revert all the transforms that affect all the channels/entities for that animation.

@prideout does that sound right to you?

**Reply by prideout** [reply]
Yes we might actually need two new methods: `resetBoneMatrices` and `resetAnimation` which will be counterparts to the existing `updateBoneMatrices` and `applyAnimation`.  Note that you will need the former if you want to go back to the T pose.

(reminder: in glTF, skinned models may or may not have animation)

**Reply by prideout** [reply]
Regarding the proposal of `resetBoneMatrices` and `resetAnimation`, the former is trivial to implement so I can go ahead and do so.

However the latter would require us to stash the original transforms of all animated nodes, which we don't currently do.  Stashing the original transform is something we should consider doing anyway, to allow for infinite animation without numeric degradation.

@starccode would `resetBoneMatrices` be sufficient for your use case?

**Reply by simonsland** [reply]
yes, I already saw the merge request, that helps a lot, thank you!

**Comment by prideout** [comment, marked as answer]
We merged #5445 for this, please open an issue if it isn't sufficient.
